explorer/types.go

```diff
@@ -218,6 +217,63 @@ type V2HostAnnouncement struct {
 	chain.V2HostAnnouncement
 }
 
+// V2FileContractResolutionType enumerates the types of file contract resolution.
+type V2FileContractResolutionType interface {
+	isV2FileContractResolution()
+}
+
+func (*V2FileContractFinalization) isV2FileContractResolution() {}
```

Comment:
Is there a reason for copying all of this from `core`? Duplication should be avoided if possible.

Reply:
In the case of renewals, we use the `V2FileContract` type for the final revision and new contract fields which gives more info than the basic type.  This is not strictly necessary but its in line with what we do everywhere else and it makes things easier for API consumers.  Because the `isV2FileContractResolution` method is unexported, we unfortunately have to copy all the other types (`V2FileContractFinalization`, `V2FileContractExpiration`, `V2StorageProof`) too.